contracts/OffchainOracle.sol

```diff
@@ -470,4 +446,45 @@ contract OffchainOracle is Ownable {
             return (true, result);
         }
     }
+
+    function _scaledMul(uint256[3] memory m, uint256 scale) private pure returns (uint256) {
+        if (m[0] == 0 || m[1] == 0 || m[2] == 0) return 0;
+
+        if (m[0] > m[1]) (m[0], m[1]) = (m[1], m[0]);
+        if (m[0] > m[2]) (m[0], m[2]) = (m[2], m[0]);
+        if (m[1] > m[2]) (m[1], m[2]) = (m[2], m[1]);
+        bool scaleApplied;
+
+        unchecked {
+            uint256 r = m[0] * m[1];
+            if (r / m[0] != m[1]) {
+                if (!_validatateMulDiv(m[0], m[1], scale)) return 0;
+                r = m[0].mulDiv(m[1], scale);
```

Comment:
Do we need check if `r == 0` here? because further division by 0 may follow

Reply:
`r` is strictly greater than 0 here because uint256 overflow happened on multiplication